src/Administration/Resources/app/administration/src/module/sw-sales-channel/view/sw-sales-channel-detail-base/index.js

```diff
@@ -611,17 +611,58 @@ export default {
         onConfirmDelete() {
             this.showDeleteModal = false;
 
-            this.$nextTick(() => {
-                this.deleteSalesChannel(this.salesChannel.id);
+            this.$nextTick(async () => {
+                const success = await this.deleteSalesChannel(this.salesChannel.id);
+
+                if (!success) {
+                    return;
+                }
+
                 this.$router.push({ name: 'sw.dashboard.index' });
             });
         },
 
         deleteSalesChannel(salesChannelId) {
-            this.salesChannelRepository.delete(salesChannelId, Context.api).then(() => {
-                Shopware.Utils.EventBus.emit('sw-sales-channel-detail-base-sales-channel-change');
-                this.salesChannelFavoritesService.refresh();
-            });
+            return this.salesChannelRepository
+                .delete(salesChannelId, Context.api)
+                .then(() => {
+                    Shopware.Utils.EventBus.emit('sw-sales-channel-detail-base-sales-channel-change');
+                    this.salesChannelFavoritesService.refresh();
+
+                    return true;
+                })
+                .catch((error) => {
+                    const current = error?.response?.data?.errors?.[0];
+                    const assignment = this.extractFkInfo(current?.detail);
+
+                    if (current?.code === '1451' && assignment) {
+                        Shopware.Store.get('error').resetApiErrors();
+                        const translated = this.$tc(`global.entities.${assignment}`, 0).toLowerCase();
+
+                        this.createNotificationError({
+                            title: this.$tc('sw-sales-channel.detail.foreignKeyDelete.title'),
```

Comment:
```suggestion
                        const translated = this.$t(`global.entities.${assignment}`, 0).toLowerCase();

                        this.createNotificationError({
                            title: this.$t('sw-sales-channel.detail.foreignKeyDelete.title'),
```

I'd prefer it to not re-introduce $tc, since it will be removed some time in the future